internal/checks/checks.go

```diff
@@ -382,6 +388,14 @@ func (c *Updater) loop(ctx context.Context) (bool, error) {
 
 	c.probe = &result.Probe
 
+	// usageReporter will be nil if the usage report configuration is disabled
+	if c.usageReporter != nil {
```

Comment:
As an alternative, not necessarily a suggestion, we could define an interface for `usage.UsageReporter` and provide two different implementations, one that pushes to Grafana usage stats, and one that is _noop_, which can be instantiated when the feature is disabled.

Reply:
I was considering what that option would look like, and personally think it's cleaner. Let me take a pass at implementing an interface and having a noop implementation 👍🏻